File: app/src/main/java/com/ellison/jetpackdemo/MyApplication.java
package com.ellison.jetpackdemo;

import android.app.Application;
import android.util.Log;

import com.ellison.jetpackdemo.lifecycle.ActivityLifecycleCallbackImpl;

import androidx.annotation.NonNull;
import androidx.camera.camera2.Camera2Config;
import androidx.camera.core.Camera;
import androidx.camera.core.CameraXConfig;
import androidx.camera.lifecycle.ProcessCameraProvider;

public class MyApplication extends Application implements CameraXConfig.Provider {
    public static final String TAG = MyApplication.class.getSimpleName();
    private ActivityLifecycleCallbacks mActivityLifecycle;

    @NonNull
    @Override
    public CameraXConfig getCameraXConfig() {
        return Camera2Config.defaultConfig();
    }

    @Override
    public void onCreate() {
        Log.d(TAG, "onCreate()");
        super.onCreate();

        mActivityLifecycle = new ActivityLifecycleCallbackImpl();
        registerActivityLifecycleCallbacks(mActivityLifecycle);
    }

    @Override
    public void onTerminate() {
        Log.d(TAG, "onTerminate()");
        super.onTerminate();
        unregisterActivityLifecycleCallbacks(mActivityLifecycle);
    }
}

File: app/src/main/java/com/ellison/jetpackdemo/MyApplication.kt
package com.ellison.jetpackdemo

import android.app.Application
import androidx.camera.core.CameraXConfig
import android.util.Log
import androidx.camera.camera2.Camera2Config
import com.ellison.jetpackdemo.lifecycle.ActivityLifecycleCallbackImpl
import dagger.hilt.android.HiltAndroidApp

@HiltAndroidApp
class MyApplication : Application(), CameraXConfig.Provider {
    private var mActivityLifecycle: ActivityLifecycleCallbacks? = null

    override fun getCameraXConfig(): CameraXConfig {
        return Camera2Config.defaultConfig()
    }

    override fun onCreate() {
        Log.d(TAG, "onCreate()")
        super.onCreate()
        mActivityLifecycle = ActivityLifecycleCallbackImpl()
        registerActivityLifecycleCallbacks(mActivityLifecycle)
    }

    override fun onTerminate() {
        Log.d(TAG, "onTerminate()")
        super.onTerminate()
        unregisterActivityLifecycleCallbacks(mActivityLifecycle)
    }

    companion object {
        @JvmField
        val TAG = MyApplication::class.java.simpleName
    }
}